caput/memh5.py

```diff
@@ -2003,10 +2008,16 @@ def _map_attr(value):
                 return value
 
         # If we are copying into memh5 ensure that any string are unicode
-        return bytes_to_unicode(value)
+        value = bytes_to_unicode(value)
+
+        # Deserialize "special" _json values
+        if key.endswith("_json"):
+            return json.loads(value)
+        else:
+            return value
 
     for key in sorted(a1):
-        a2[key] = _map_attr(a1[key])
+        a2[key] = _map_attr(key, a1[key])
```

Comment:
Should the keys get renamed in here? i.e. `version_json` -> `version` when being read (and vice-versa),

Reply:
Then it wouldn't get serialized again if it's written to hdf5 later.

Reply:
I think the solution to that is fairly easy, you just do:
```python
if isinstance(a1[key], (dict, list, tuple)):
    a2[key + "_json"] = json.dumps(a1[key])
```

Reply:
The rationale being that `h5py` can't write those types anyway, so it doesn't matter if you do something special with them.

Reply:
We can do that, but only for `dict`'s, since `tuple`'s and `list`'s are converted to numpy arrays in `_map_attr`.

Reply:
Hmmm. Good point. Then maybe we just convert dicts. I think that's mostly fine.

Reply:
It might be cleaner to not change the key names, but have the serialised value start with a magic string. Something like: `!!json:{"an_example_json": "dict"}`

Reply:
That seems less likely to give any potential clashes to me and unexpected behaviour to me.

Reply:
Maybe even: `!!_memh5_json:{"an_example_json": "dict"}`